scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -297,14 +297,27 @@ def _str2bool(v: str) -> bool:
                             run_command(cmd, failure_list, debug_mode)
                 except Exception as e:
                     logger.log_error(f"Exception in loading model config: {str(e)}")
-                    failure_list.append(asset.name)
+                    failure_list.append(asset)
 
             else:
                 logger.log_warning(f"unsupported asset type: {asset.type.value}")
 
     if len(failure_list) > 0:
-        logger.log_warning(
-            f"following assets failed to publish: {failure_list}")
+        failed_assets = defaultdict(list)
+        logger.print(f"failure_list: {failure_list}")
+        for asset in failure_list:
+            logger.print(f"processing failed asset: {asset.name}")
+            failed_assets[asset.type.value].append(asset.name)
+
+        logger.print(f"failed assets: {failed_assets}")
+        for asset_type, asset_names in failed_assets.items():
+            logger.log_warning(f"Failed to register {asset_type}s: {asset_names}")
+        # the following dump process will generate a yaml file for the report
+        # process in the end of the publishing script
+        result_path = Path().resolve() / "failed_assets.yml"
+        dump_data = {asset_type: asset_names for asset_type, asset_names in failed_assets.items()}
+        with open(result_path, "w") as file:
+            yaml.dump(dump_data, file, default_flow_style=False, sort_keys=False)
```

Comment:
I think we need to cut down on verbosity and this should be simpler:
```suggestion
        for asset in failure_list:
            failed_assets[asset.type.value].append(asset.name)

        for asset_type, asset_names in failed_assets.items():
            logger.log_warning(f"Failed to register {asset_type}s: {asset_names}")
        # the following dump process will generate a yaml file for the report
        # process in the end of the publishing script
        with open("failed_assets.yml", "w") as file:
            yaml.dump(failed_assets, file, default_flow_style=False, sort_keys=False)
```

Reply:
The reason why I have `dump_data` is that `yaml.dump` a `defaultdict(list)` will give us a yaml file beginning with 
!!python/object/apply:collections.defaultdict
args:
- !!python/name:builtins.list ''
dictitems:

And `dump_data` could eliminate these unwanted lines. What do you think?

Reply:
You could also just do `yaml.dump(dict(failed_assets), file, default_flow_style=False, sort_keys=False)` ([reference](https://stackoverflow.com/questions/20428636/how-to-convert-defaultdict-to-dict))